pkg/cmd/container/create.go

```diff
@@ -357,8 +366,47 @@ func generateRootfsOpts(args []string, id string, ensured *imgutil.EnsuredImage,
 			// error message is from Podman
 			return nil, nil, errors.New("no command or entrypoint provided, and no CMD or ENTRYPOINT from image")
 		}
+
+		entrypointPath = processArgs[0]
+
 		opts = append(opts, oci.WithProcessArgs(processArgs...))
 	}
+
+	isEntryPointSystemd := (entrypointPath == "/sbin/init" ||
+		entrypointPath == "/usr/sbin/init" ||
+		entrypointPath == "/usr/local/sbin/init")
+
+	stopSignal := options.StopSignal
+
+	if options.Systemd == "always" || (options.Systemd == "true" && isEntryPointSystemd) {
+		if options.Privileged {
+			securityOptsMap := strutil.ConvertKVStringsToMap(strutil.DedupeStrSlice(options.SecurityOpt))
+			privilegedWithoutHostDevices, err := maputil.MapBoolValueAsOpt(securityOptsMap, "privileged-without-host-devices")
+			if err != nil {
+				return nil, nil, err
+			}
+
+			// See: https://github.com/containers/podman/issues/15878
+			if !privilegedWithoutHostDevices {
+				return nil, nil, errors.New("if --privileged is used with systemd `--security-opt privileged-without-host-devices` must also be used")
+			}
+		}
+
+		opts = append(opts,
+			oci.WithoutMounts("/sys/fs/cgroup"),
```

Comment:
@sazzy4o I found your change while looking into supporting containers with systemd using k8s + container. I did the tmpfs mounts for /run, /tmp/, etc., but I was mounting the host's /sys/fs/cgroup as ready-only, which didn't work.

Here, you are removing the mount, which caught my attention. Is this so that systemd creates /sys/fs/cgroup when it initializes?

Reply:
@jfernandez Yes, this allow systemd to run inside the container and create `/sys/fs/cgroup`

This was based on the [podman --systemd flag](https://docs.podman.io/en/latest/markdown/podman-run.1.html#systemd-true-false-always)